### DISCUSSION TITLE
Does Budybase have a "subscription payment wall’ feature?

### DISCUSSION THREAD

**Discussion by Rich3art** [Budibase/budibase#7798] [Help] [answered]
Hi Everyone,

I have two questions here please:

1. I just would like to please find out does Budybase have a "subscription payment wall’ feature that I can implement before an account is registered on my live App website that a payment amount is made, which is then a subscription on the same day as signing up. This will then check every time if the subscription is still active and prevent login access to the site unless the subscription is active.
(As an example more or less the same as what the plugin “Membership pro” on “WordPress” does if anyone is familiar with Wordpress)

2. Also does Budibase have a feature where everybody can register as a basic user as default, and then it will require me to change their user access if I desire so or I put something in place that will automatically change their type of access once they click on a button and "request access from me" 

If neither of these then how can I implement this on Budibase or inside Budibase?

The only option I know of is running WordPress in parallel with this but it will really badly affect my server load and don’t want to do this & I'm also not 100% sure how to do this either.

I'm not a skilled developer but I’m learning as I go, please help, would really appreciate it.

Kind Regards,
Richard

**Comment by melohagan** [comment, marked as answer]
Hi @Rich3art 

1. Budibase does not have a "subscription payment wall" feature. It may be possible to use the REST connector to build an app that integrates with Stripe, etc, but there's nothing native.

2. An admin user can navigate to the **Users** tab in the portal. When you click `Add users` you can select *App user* as the role, which is basic.  
![Screenshot 2022-09-16 at 08 24 03](https://user-images.githubusercontent.com/101575380/190580739-9a862ec9-98a7-4265-8ae3-5f8977a43044.png)

![Screenshot 2022-09-16 at 08 22 47](https://user-images.githubusercontent.com/101575380/190580517-6892a832-1689-49b4-8a3c-c1369b6df7a9.png)

Or where you looking more a system of self sign up?

**Reply by Rich3art** [reply]
Hi Melohagan,

Thanks for the reply, yes I was looking basically for a system that automates the whole process.

I.o.w you as the user go to the website URL and you are interested to use the program but before you have access to do so you need to pay a monthly subscription that will then automatically upgrade your access to be able to use whatever app or apps you have paid for on the subscription.

I had a look at Buddybase and so far it meets all my needs except this one small thing.

How do I build an app that integrates with Stripe, etc using the REST connector, is there perhaps one that's already built that I can just modify or can you perhaps direct me to a tutorial because I am not skilled in code just on minimal low code?

Kind Regards,
Richard

**Reply by melohagan** [reply]
We're soon to roll out custom datasource integrations that will over time greatly expand the selection of API connectors that Budibase has to offer.
I'll work on the Stripe connector next!

**Reply by melohagan** [reply]
Plugin created for handling Stripe Subscriptions, Subscription Items, Subscription Schedules: https://github.com/melohagan/stripe-subscriptions-custom-datasource

There's a lot to Stripe, so if you find you need access to another set of endpoints then you can let me know, but I'm not in a position myself to guide you through the process of setting up a stripe payment flow!
